From cfd3fe9704d72c776bf86117b9517d61e2e738e7 Mon Sep 17 00:00:00 2001
From: Arnaud GRANAL <arnaud@geometry.ee>
Date: Sat, 11 Dec 2021 06:29:38 +0000
Subject: [PATCH 456/506] Declare user-agent as Windows on specific websites
 that forces users to download native app despite being compatible with Kiwi

---
 net/url_request/url_request_http_job.cc | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/net/url_request/url_request_http_job.cc b/net/url_request/url_request_http_job.cc
index 085847a9..4ed7b1ac 100644
--- a/net/url_request/url_request_http_job.cc
+++ b/net/url_request/url_request_http_job.cc
@@ -343,9 +343,12 @@ void URLRequestHttpJob::OnGotFirstPartySetMetadata(
   {
      request_info_.extra_headers.SetHeader(HttpRequestHeaders::kUserAgent, "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0");
   }
-  else if (request_info_.url.host().find("facebook.com") != std::string::npos)
+  else if (request_info_.url.host().find("roninchain.com") != std::string::npos || request_info_.url.host().find("marketplace.axieinfinity.com") != std::string::npos)
   {
-     request_info_.extra_headers.SetHeader(HttpRequestHeaders::kUserAgent, "Mozilla/5.0 (Mobile; rv:48.0; A405DL) Gecko/48.0 Firefox/48.0 KAIOS/2.5");
+     request_info_.extra_headers.SetHeader(HttpRequestHeaders::kUserAgent, "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.45 Safari/537.36");
+     request_info_.extra_headers.SetHeader("Sec-CH-UA", "\" Not A;Brand\";v=\"99\", \"Chromium\";v=\"96\", \"Google Chrome\";v=\"96\"");
+     request_info_.extra_headers.SetHeader("Sec-CH-UA-Mobile", "?0");
+     request_info_.extra_headers.SetHeader("Sec-CH-UA-Platform", "Windows");
   }
   else if (request_info_.url.host().find("news.google.com") != std::string::npos && request_info_.url.path().find("CAAqJggKIiBDQkFTRWdvSUwyMHZNRFZxYUdjU0FtVnVHZ0pWVXlnQVAB") != std::string::npos)
   {
